src/mono/sample/wasm/browser-eventpipe/Program.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Runtime.InteropServices.JavaScript;
+using System.Runtime.InteropServices;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+
+namespace Sample
+{
+    class ConsoleWriterEventListener : EventListener
+    {
+        public static ConsoleWriterEventListener Instance;
+
+        protected override void OnEventSourceCreated(EventSource eventSource)
+        {
+            if(eventSource.Name == "WasmHello")
+            {
+                EnableEvents(eventSource, EventLevel.Informational);
+            }
+            if(eventSource.Name == "System.Runtime")
+            {
+                EnableEvents(eventSource, EventLevel.Informational);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            Console.WriteLine(eventData.TimeStamp + " " + eventData.EventName);
+        }
+    }
+
+    public partial class Test
+    {
+        public static int Main(string[] args)
+        {
+            // ActivityTracker.Instance.Enable(); via reflection
+            var type = typeof(EventSource).Assembly.GetType("System.Diagnostics.Tracing.ActivityTracker");
+            var prop = type.GetProperty("Instance", BindingFlags.Static | BindingFlags.Public);
+            var m = type.GetMethod("Enable");
+            var instance = prop.GetValue(null);
+            m.Invoke(instance, null);
```

Comment:
Are we doing this for any concrete constraint on the repo or is this a limitation that will also impact customers.

Reply:
Thanks for asking, I forgot about `ActivityTracker.Instance.Enable()` mess. 
We discussed it more [here](https://github.com/dotnet/runtime/issues/56073#issuecomment-2721399456)

CoreCLR just always does it. In mono it's very convoluted.
I think I will just always call it from some CoreLib place when the feature is enabled. Similar to CoreCLR

Reply:
I'll do it here https://github.com/dotnet/runtime/pull/114326 and remove this hack